src/components/CreatePostInput/index.js

```diff
@@ -132,51 +129,9 @@ const CreatePostInput = ({
     const isHashtagDetected = v.includes('#') && positionHashtag > positionMention;
 
     if (isHashtagDetected) {
-      const spaceStatus = v.includes(' ', positionHashtag);
-      const detectEnter = v.includes('\n', positionHashtag);
-      const textSearch = v.substring(positionHashtag + 1);
-      setHashtagPosition(positionHashtag);
-      /**
-       * cari posisi kursor dimana
-       * cek apakah posisi sebelum kursor # atau bukan
-       * ambil semua value setelah posisi #
-       */
-      if (!spaceStatus && !detectEnter) {
-        setPositionTopicSearch(positionHashtag);
-        searchTopicDebounced(textSearch);
-        setPositionKeyboard('always');
-        setLastTopicInCursor(textSearch);
-      } else {
-        resetTopicSearch();
-        setPositionKeyboard('never');
-        cancelAllRequest();
-        const removeCharacterAfterSpace = textSearch.split(' ')[0];
-        const hashtagLastCharIndex = positionHashtag + removeCharacterAfterSpace?.length;
-        if (hashtagLastCharIndex === positionEndCursor - 1) {
-          const newTopics = joinTopicIntoTopicList(removeCharacterAfterSpace, topics);
-          setTopics(newTopics);
-          // setHashtags(newTopics)
-        }
-
-        setLastTopicInCursor('');
-      }
-
-      // handleStateHashtag(v, setTopics, positionEndCursor);
+      handleHashtagDetection(v, positionHashtag);
     } else if (isMentionDetected) {
-      const spaceStatus = v.includes(' ', positionMention);
-      const detectEnter = v.includes('\n', positionMention);
-      const textSearch = v.substring(positionMention + 1);
-      setHashtagPosition(positionMention);
-      if (!spaceStatus && !detectEnter) {
-        setPositionTopicSearch(positionMention);
-        searchUserDebounced(textSearch);
-        setPositionKeyboard('always');
-      } else {
-        resetListUsersForTagging();
-        setPositionKeyboard('never');
-        cancelAllRequest();
-      }
-      // handleStateMention(v);
+      handleMentionDetection(v, positionMention);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Refactoring the code for detecting hashtags and mentions into separate functions improves readability and maintainability. Good job on this change!
```diff
+ handleHashtagDetection(v, positionHashtag);
+ handleMentionDetection(v, positionMention);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->